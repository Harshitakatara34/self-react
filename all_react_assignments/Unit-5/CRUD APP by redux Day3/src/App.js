import logo from './logo.svg';
import './App.css';
import { Provider } from 'react';
import { Login } from './Pages/Login';
import Navbar from './Components/Navbar';
import MainRoutes from './Pages/MainRoutes';
function App() {
  return (
    <div className="App">
    <Navbar/>
   <MainRoutes/>
    </div>
  );
}

export default App;
