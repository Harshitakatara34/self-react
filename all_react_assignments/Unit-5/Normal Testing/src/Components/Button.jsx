import React from 'react'
import styles from "./button.module.css"
export const Button = ({children,color,size,handleAdd}) => {
  return (
    <button onClick={handleAdd} data-testid="custom-button" className={`${styles[color]} ${styles[size]}`}>{children}</button>
  )
}