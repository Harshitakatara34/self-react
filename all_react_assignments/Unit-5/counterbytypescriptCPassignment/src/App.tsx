

import React, { useState, useEffect } from "react";

function App() {
  const [count, setCount] = useState<number>(0);

  useEffect(() => {
    // Get the counter value from localStorage when the component mounts
    const storedCount = localStorage.getItem("counter");
    if (storedCount !== null) {
      setCount(parseInt(storedCount));
    }
  }, []);

  useEffect(() => {
    // Save the counter value to localStorage whenever it changes
    localStorage.setItem("counter", count.toString());
  }, [count]);

  const handleIncrement = () => {
    setCount(count + 1);
  };

  const handleDecrement = () => {
    setCount(count - 1);
  };

  const handleIncrement5 = () => {
    setCount(count + 5);
  };

  const handleDecrement5 = () => {
    setCount(count - 5);
  };

  return (
    <div className="App">
      <h1 data-testid="counter">{count}</h1>
      <button data-testid="increment" onClick={handleIncrement}>
        Increment
      </button>
      <button data-testid="decrement" onClick={handleDecrement}>
        Decrement
      </button>
      <button data-testid="increment5" onClick={handleIncrement5}>
        Increment 5 times
      </button>
      <button data-testid="decrement5" onClick={handleDecrement5}>
        Decrement 5 times
      </button>
    </div>
  );
}

export default App;
