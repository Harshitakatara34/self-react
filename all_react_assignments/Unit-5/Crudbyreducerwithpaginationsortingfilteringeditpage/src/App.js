import logo from './logo.svg';
import './App.css';
import { Provider } from 'react';
import { Login } from './Pages/Login';
import Navbar from './Components/Navbar';
import MainRoutes from './Pages/MainRoutes';
import Pagination from './Components/Pagination';
function App() {
  return (
    <div className="App">
    <Navbar/>
   <MainRoutes/>
   
    </div>
  );
}

export default App;
