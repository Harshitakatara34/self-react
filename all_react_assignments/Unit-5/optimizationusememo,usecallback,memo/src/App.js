import logo from './logo.svg';
import './App.css';
import TododInput from './Components/TododInput';
function App() {
  return (
    <div className="App">
      <TododInput/>
    </div>
  );
}

export default App;
