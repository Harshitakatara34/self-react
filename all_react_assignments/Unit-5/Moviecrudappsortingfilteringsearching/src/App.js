import "./App.css";
import { MovieList } from "./Components/MovieList";
import { Navbar } from "./Components/Navbar";
import { Login } from "./Pages/Login";
import { MainRoutes } from "./Pages/MainRoutes";

function App() {
  //Do not modify anything in this file
  return (
    <div className="App">
      <Navbar />
      <MainRoutes />
     
      
    </div>
  );
}

export default App;
