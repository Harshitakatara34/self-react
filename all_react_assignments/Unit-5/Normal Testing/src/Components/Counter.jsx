import React, { useState } from 'react'
import { Button } from './Button'

export const Counter = () => {
  const [count,setCount]=useState(0)
  const handleAdd=()=>{
    setCount(count+1)
  }
    return (
    <div data-testid="counter"><h2>counter: {count}</h2>
    {/* <button data-testid="add-btn" 
    onClick={handleAdd}>Click Me</button> */}
    <Button color={"text"} size={"small"} handleAdd={handleAdd}>Add</Button>
    </div>
  )
}