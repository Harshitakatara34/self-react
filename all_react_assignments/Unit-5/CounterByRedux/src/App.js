import logo from './logo.svg';
import './App.css';
import { Todo } from './Components/Todos';
import Counter from './Components/Counter';
function App() {
  return (
    <div className="App">
   <Counter/>
   <Todo/>
    </div>
  );
}

export default App;
