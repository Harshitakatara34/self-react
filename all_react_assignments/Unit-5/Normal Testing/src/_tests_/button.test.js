import App from "../App";
import {fireEvent, render,screen} from "@testing-library/react";
import { Button } from "../Components/Button"
import renderer from "react-test-renderer"

describe("Testing Button component",()=>{
    it("Buttons should render on DOM",()=>{
        render (
       <Button size={"large"} color={"teal"}>
        Click Me</Button>
            )
        const button=screen.getByText("Click Me")
        expect(button).toBeInTheDocument()
            // screen.debug();
    })
    it("Buttons should be present on DOM",()=>{
        render (
       <Button>
      Custom Button</Button>
            )
        const button=screen.getByTestId("custom-button")
        expect(button).toBeInTheDocument()
            // screen.debug();
    })
    it("Check for button text content",()=>{
        render(<App/>)
        const button=screen.getAllByTestId("custom-button")
        expect(button[0]).toHaveTextContent("Click Me")
        expect(button[1]).toHaveTextContent("Click")
    })
   it("should incoke the function",()=>{
const mockFunc=jest.fn();
    render(<Button handleAdd={mockFunc}/>)
    const button=screen.getByTestId("custom-button")
fireEvent.click(button)
expect(mockFunc).toBeCalledTimes(1)
})
    

//snapShot testing
it("should match the snapshot",()=>{
const tree=renderer.create(<Button color={"blue"} size={"large"} >Click Me</Button>).toJSON()
expect(tree).toMatchSnapshot();


})

   
})