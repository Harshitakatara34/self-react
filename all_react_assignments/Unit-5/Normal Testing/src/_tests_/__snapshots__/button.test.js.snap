// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button component should match the snapshot 1`] = `
<button
  className="blue large"
  data-testid="custom-button"
>
  Click Me
</button>
`;
